src/Features/CSharp/Portable/RemoveUnusedParametersAndValues/CSharpRemoveUnusedParametersAndValuesCodeFixProvider.cs

```diff
@@ -0,0 +1,187 @@
+                    var identifierName = (IdentifierNameSyntax)node;
+                    return identifierName.WithIdentifier(newName.WithTriviaFrom(identifierName.Identifier));
+
+                case SyntaxKind.VariableDeclarator:
+                    var variableDeclarator = (VariableDeclaratorSyntax)node;
+                    return variableDeclarator.WithIdentifier(newName.WithTriviaFrom(variableDeclarator.Identifier));
+
+                case SyntaxKind.SingleVariableDesignation:
+                    return SyntaxFactory.SingleVariableDesignation(newName).WithTriviaFrom(node);
+
+                case SyntaxKind.CatchDeclaration:
+                    var catchDeclaration = (CatchDeclarationSyntax)node;
+                    return catchDeclaration.WithIdentifier(newName.WithTriviaFrom(catchDeclaration.Identifier));
+
+                default:
+                    throw ExceptionUtilities.Unreachable;
+            }
+        }
+
+        protected override ILocalSymbol GetSingleDeclaredLocal(LocalDeclarationStatementSyntax localDeclaration, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            Contract.ThrowIfFalse(localDeclaration.Declaration.Variables.Count == 1);
+            return (ILocalSymbol)semanticModel.GetDeclaredSymbol(localDeclaration.Declaration.Variables[0]);
+        }
+
+        protected override void InsertAtStartOfSwitchCaseBlock(SwitchSectionSyntax switchCaseBlock, SyntaxEditor editor, LocalDeclarationStatementSyntax declarationStatement)
+        {
+            var firstStatement = switchCaseBlock.Statements.FirstOrDefault();
+            if (firstStatement != null)
+            {
+                editor.InsertBefore(firstStatement, declarationStatement);
+            }
+            else
+            {
+                // Switch section without any statements is an error case.
+                // Insert before containing switch statement.
+                editor.InsertBefore(switchCaseBlock.Parent, declarationStatement);
+            }
+        }
+
+        protected override Task<SyntaxNode> RemoveDiscardDeclarationsAsync(
+            SyntaxNode memberDeclaration,
+            Document document,
+            CancellationToken cancellationToken)
+        {
+            var editor = new SyntaxEditor(memberDeclaration, SyntaxGenerator.GetGenerator(document));
+            foreach (var child in memberDeclaration.DescendantNodes(descendIntoChildren: n => !(n is ExpressionSyntax)))
+            {
+                if (child is LocalDeclarationStatementSyntax localDeclarationStatement &&
+                    localDeclarationStatement.Declaration.Variables.Any(v => v.Identifier.Text == "_"))
+                {
+                    ProcessVariableDeclarationWithDiscard(localDeclarationStatement, editor);
+                }
+                else if (child is CatchDeclarationSyntax catchDeclaration &&
+                    catchDeclaration.Identifier.Text == "_")
```

Comment:
continued use of constant.  Please extract out some helper.  I'd even be ok with a SyntaxToken or String extension for this.